File: aqefrontend/src/app/hr-module/model/topic.ts
import { Question } from './question';

export class Topic {
    id: String;
    name: String;
    questions: Question[];
    createdAt: Date;
    link: String;
    keywords: String[];
    constructor(i,n,q,c,l,k){
        this.id=i;this.name=n;this.questions=q;this.createdAt=c;this.link=l;this.keywords=k;
    }
}


// private String id;
// private String name;
// private List<Question> questions;
// private Date createdAt;
// private String link;
// private List<String> keywords;

File: hrService/src/main/java/com/stackroute/hrservice/model/Topic.java
package com.stackroute.hrservice.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@Data
@Document
public class Topic {
  @Id
  private String id;
  private String name;
  private String createdAt;
  private String link;
  private List<String> keywords;

  public Topic() {
  }

  public Topic(String id, String name, String createdAt, String link, List<String> keywords) {
    this.id = id;
    this.name = name;
    this.createdAt = createdAt;
    this.link = link;
    this.keywords = keywords;
  }
}
